torch/_inductor/scheduler.py

```diff
@@ -4049,15 +4099,34 @@ def get_graph_partition_signature(
                 for name in partition_input_names
                 if name in name_to_node
             }
+
+            # returns an input tensor as output if it is not freed. This allows
+            # more cudagraph managed tensors and is important for handling saved
+            # tensors.
+            extra_output_names = [
+                name
+                for name in partition_input_names
+                if name in name_to_node and name not in buffer_names_to_free
+            ]
+
+            returned_output_names.update(extra_output_names)
+
             output_nodes = [name_to_node[name] for name in returned_output_names]
-            signatures.append(
-                GraphPartitionSignature(
-                    input_nodes,
-                    output_nodes,
-                    input_deallocation,
-                    skip_cudagraph,
-                )
+
+            constant_names = [
+                name for name in partition_input_names if name not in name_to_node
+            ]
```

Comment:
This seems like a weird way of getting constants.. is there another way of doing this ? or could you say more about this

Reply:
Could we use constant names instead ?